runtime/Cpp/runtime/src/atn/SemanticContext.cpp

```diff
@@ -208,35 +241,31 @@ std::string SemanticContext::AND::toString() const {
 //------------------ OR ------------------------------------------------------------------------------------------------
 
 SemanticContext::OR::OR(Ref<const SemanticContext> const& a, Ref<const SemanticContext> const& b) {
-  Set operands;
+  std::unordered_set<const SemanticContext*, SemanticContextHasher, SemanticContextComparer> operands;
+  Ref<const SemanticContext::PrecedencePredicate> precedencePredicate;
 
   if (is<const OR>(a)) {
     for (const auto &operand : std::dynamic_pointer_cast<const OR>(a)->opnds) {
-      operands.insert(operand);
+      insertSemanticContext<std::greater<int>>(operand, operands, opnds, precedencePredicate);
     }
   } else {
-    operands.insert(a);
+    insertSemanticContext<std::greater<int>>(a, operands, opnds, precedencePredicate);
   }
 
   if (is<const OR>(b)) {
     for (const auto &operand : std::dynamic_pointer_cast<const OR>(b)->opnds) {
-      operands.insert(operand);
+      insertSemanticContext<std::greater<int>>(operand, operands, opnds, precedencePredicate);
     }
   } else {
-    operands.insert(b);
+    insertSemanticContext<std::greater<int>>(b, operands, opnds, precedencePredicate);
   }
 
-  std::vector<Ref<const PrecedencePredicate>> precedencePredicates = filterPrecedencePredicates(operands);
-  if (!precedencePredicates.empty()) {
-    // interested in the transition with the highest precedence
```

Comment:
Please keep this comment. There's not so much description in the code, so every line counts :-)

Reply:
Added back.